src/main/groovy/edu/oregonstate/mist/musicapi/resources/ShelfResource.groovy

```diff
@@ -0,0 +1,126 @@
+package edu.oregonstate.mist.musicapi.resources
+
+import edu.oregonstate.mist.api.Resource
+import edu.oregonstate.mist.musicapi.core.Album
+import edu.oregonstate.mist.musicapi.core.Shelf
+import edu.oregonstate.mist.api.AuthenticatedUser
+import io.dropwizard.auth.Auth
+import javax.ws.rs.GET
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.Consumes
+import javax.ws.rs.core.Context
+import javax.ws.rs.core.Response
+import javax.ws.rs.core.UriBuilder
+import javax.ws.rs.core.UriInfo
+
+import org.skife.jdbi.v2.DBI
+import org.skife.jdbi.v2.util.IntegerMapper
+
+/**
+ * Shelf resource class.
+ */
+@Path('/shelf')
+@groovy.transform.TypeChecked
+class ShelfResource extends Resource {
+    private DBI dbi
+
+    @Context
+    private UriInfo uriInfo
+
+    public ShelfResource(DBI dbi) {
+        this.dbi = dbi
+    }
+
+    @POST
+    @Consumes("application/json")
+    @Produces("application/json")
+    Response createShelf(@Auth AuthenticatedUser _, @PathParam("id") int id, Shelf shelf) {
+        def h = this.dbi.open()
+        try {
+            def q = h.createStatement("INSERT INTO MUS_SHELF (id, name, created) VALUES (NULL, ?, SYSDATE)")
+            q.bind(0, shelf.name)
+            def gk = q.executeAndReturnGeneratedKeys()
+            def rowid = gk.first().get("rowid")
+
+            q = h.createQuery("SELECT * FROM MUS_SHELF WHERE rowid = ?")
+            q.bind(0, rowid)
+            shelf = q.map(Shelf).first()
+            shelf.albumUrls = []
+            return this.created(shelf).build()
+        } finally {
+            h.close()
+        }
+    }
+
+    /**
+     * Responds to GET requests by returning a message.
+     *
+     * @return message
+     */
+    @GET
+    @Path('{id}')
+    @Produces("application/json")
+    Response getShelf(@Auth AuthenticatedUser _, @PathParam("id") int id) {
+        def h = this.dbi.open()
+        try {
+            // Get the shelf
+            Shelf shelf
+            def q = h.createQuery('''
+                SELECT id, name,
+                    to_char(created, 'YYYY-MM-DD"T"HH24:MI:SS"Z"') as created
+                FROM MUS_SHELF
+                WHERE id = ?
+            ''')
+            q.bind(0, id)
+            shelf = q.map(Shelf).first()
+            if (shelf == null) {
```

Comment:
I could, but i'd rather be explicit. I don't care about whether shelf is truthy, i care about whether it is null, because that's the error condition. If i use !shelf and shelf gains an asBoolean method, this check would become subtly wrong.